### PR TITLE
Replace custom Base32 encoding

### PR DESCRIPTION
**Pull request by hennekey**
Guava is available so a custom implementation is unnecessary.

### CODE REVIEW

File: commons/src/main/java/org/archive/util/Base32.java [hunk 1]
```diff
@@ -18,142 +18,25 @@
...
-                digit = (currByte >> (8 - (index + 5))) & 0x1F;
-                index = (index + 5) % 8;
-                if (index == 0)
-                    i++;
-            }
-            base32.append(base32Chars.charAt(digit));
-        }
-
-        return base32.toString();
+      return BaseEncoding.base32().encode(bytes);
```
Comment (ato) [inline commons/src/main/java/org/archive/util/Base32.java]: There seems to be an output change here:

original: `Base32.encode("a") => "ME"`
new: `Base32.encode("a") => "ME======"`

We might need `omitPadding()` to ensure this is an exact drop in replacement.

File: commons/src/main/java/org/archive/util/Base32.java [hunk 2]
```diff
@@ -18,142 +18,25 @@
...
-            if (b < 0) {
-                b += 256;
-            }
-            System.out.print((Integer.toHexString(b + 256)).substring(1));
-        }
-        System.out.println();
-        System.out.println("Reencoded: " + Base32.encode(decoded));
+    @Deprecated
+    static public byte[] decode(final String base32)  {
+      return BaseEncoding.base32().decode(base32);
```
Comment (ato) [inline commons/src/main/java/org/archive/util/Base32.java]: This is throwing an exception on lowercase input such as in `ContentDigestHistoryTest.testBasics()`.

Comment (hennekey) [reply commons/src/main/java/org/archive/util/Base32.java]: Got it.

**Comment by ato** [comment]
Thanks!

**Pull request merged**